---
title:               "BAG3 Pathway Analysis"
author:
  - name:            "Emir Turkes"
  - name:            "Columbia University"
date:                '`r strftime(Sys.time(), format = "%B %d, %Y")`'
bibliography:        "../bag3-pathway-analysis.bib"
biblio-style:        apalike
link-citations:      true
output:
  html_document:
    number_sections: true
    theme:           lumen
    toc:             true
    toc_depth:       2
    toc_float:
      collapsed:     false
      smooth_scroll: false
knit:
  (function(inputFile, encoding) {
    rmarkdown::render(
      inputFile, encoding = encoding, output_file = "../results/bag3-pathway-analysis-report.html")})
---

```{r, include = FALSE}
# Copyright 2019 Emir Turkes
# 
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
# 
#     http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

knitr::opts_chunk$set(echo = TRUE, fig.width = 8.5)
```

<style type="text/css">
body {font-size: 16px;}
h1.title {font-size: 35px;}
h1 {font-size: 24px;}
h2 {font-size: 22px;}
h3 {font-size: 20px;}
.toc-content {padding-left: 0px; padding-right: 0px;}
div.tocify {width: 100%;}
.tocify-subheader .tocify-item {font-size: 0.95em; padding-left: 25px; text-indent: 0;}
div.main-container {max-width: none; width: 100%;}
</style>

*This is an in-depth analysis based on one or more initial broad analyses.*

The background for this data is as follows:

- Co-IP MS (co-immunoprecipitation mass spectrometry) using BAG3 as bait from [github.com/eturkes/bag3-co-ip](https://github.com/eturkes/bag3-co-ip).

This analysis was performed in R except where noted.
The source code and instructions for rerunning the analysis can be found at [github.com/eturkes/bag3-pathway-analysis](https://github.com/eturkes/bag3-pathway-analysis).

# Final Results

**Read just this section for the final results of the analysis and a summary of the methods.**

# ~~~~ Breakdown of Methods ~~~~ {-}

The following sections breakdown the methods used to transform the contents in `Original Data` to those in `Final Results`.
We start by loading in any required packages for our analysis and setting some global variables.

```{r}
library(conflicted)
library(devtools)
library(DT)
library(gprofiler2)
```

```{r}
assets_dir <- file.path(getwd(), "..", "assets")
results_dir <- file.path(getwd(), "..", "results")
```

# Original Data

This section contains the cleaned dataset generated from the analysis in [github.com/eturkes/bag3-co-ip](https://github.com/eturkes/bag3-co-ip).

```{r}
co_ip_genes <- readRDS(file.path(assets_dir, "bag3-co-ip", "co_ip_genes_final.rds"))
cols <- c("Genes", "Specificity: avg(Abun. Ratio, Watson)", "Abundance Ratio: (scr/sh)",
          "Watson Confidence", "scrRNA Abundance", "shBAG3 Abundance")
```

## Sort Schemes {.tabset}

Two sort schemes are presented, one to show specificity with BAG3 and another to show the most abundant Co-IP proteins.

### Specificity with BAG3

```{r}
co_ip_genes_spec <- co_ip_genes[order(-co_ip_genes[[cols[2]]], co_ip_genes[[cols[6]]]), ]
rownames(co_ip_genes_spec) <- NULL # Reset row numbers.
eturkesRutils::datatable(co_ip_genes_spec) %>% # From my own package, provides custom formatting.
  formatRound(c(2, 3, 4)) %>% formatStyle(cols[1], `text-align` = "center")
```

### Abundance of Protein

```{r}
co_ip_genes_abun <- co_ip_genes[order(-co_ip_genes[[cols[5]]], -co_ip_genes[[cols[6]]]), ]
rownames(co_ip_genes_abun) <- NULL # Reset row numbers.
eturkesRutils::datatable(co_ip_genes_abun) %>% # From my own package, provides custom formatting.
  formatRound(c(2, 3, 4)) %>% formatStyle(cols[1], `text-align` = "center")
```

# g:Profiler

We start by taking our gene lists and inputing them into [g:Profiler](https://biit.cs.ut.ee/gprofiler/gost), a web service for functional enrichment analysis [@raudvere_g:profiler:_2019].
The tool can be operated interactively through a web browser, but an R client is also provided, so we use that.
g:Profiler is the first step in a 2019 Nature Protocols paper on pathway analysis [@reimand_pathway_2019] that we will generally be following for this analysis.

## Sort Schemes {.tabset}

Since we have two gene lists, we shall continue analyzing them in parallel.

### Specificity with BAG3

```{r}
gprofiler_spec <- gost(co_ip_genes_spec$Genes, organism = "rnorvegicus", ordered_query = TRUE)
gostplot(gprofiler_spec)
```

```{r, fig.width = 11, fig.height = 93}
publish_gosttable(gprofiler_spec)
```

### Abundance of Protein

```{r}
gprofiler_abun <- gost(co_ip_genes_abun$Genes, organism = "rnorvegicus", ordered_query = TRUE)
gostplot(gprofiler_abun)
```

```{r, fig.width = 11, fig.height = 94}
publish_gosttable(gprofiler_abun)
```

# GSEA

GSEA (gene set enrichment analysis) [@subramanian_gene_2005], is a common pathway analysis method when dealing with a ranked gene list. However, this method is only suitable when near-genomewide expression profiles are available, so we hold off using it until data such as an RNA-Seq set is ready.

# Cytoscape

Cytoscape [@shannon_cytoscape:_2003] is an open-source platform for visualization of networks with an extensive collection of third-party apps.
Together with the apps Enrichment Map [@merico_enrichment_2010] and AutoAnnotate [@kucera_autoannotate:_2016], results from g:Profiler and GSEA can be visualized in an intuitive way.
Although APIs exist to carry out this process programmatically, it is unwieldy considering that networks must be manually adjusted in almost all cases, so at least for the time being we opt to use the Cytoscape GUI.

![gprofiler-bag3-co-ip.svg](../assets/gprofiler/gprofiler-bag3-co-ip.svg "BAG3 Co-IP Specificity vs. Abundance")

# Conclusion

This concludes the methods used in this analysis.
Below we simply write the data to disk for exploration in `Final Results` at the top of this report and output the `session_info`.

```{r}
session_info()
```

# References
